### PR TITLE
Cleaned Up Outlook Commands

### PR DESCRIPTION
**Pull request by openbots-ff**

### CODE REVIEW

**Review by CameronHerwig — DISMISSED** [review]
All looks good to me.

File: taskt/Core/Automation/Commands/Outlook Commands/GetOutlookEmailsCommand.cs [hunk 1]
```diff
@@ -1,109 +1,115 @@
...
         [XmlAttribute]
-        [Attributes.PropertyAttributes.PropertyDescription("Assign MailItem List to variable")]
-        [Attributes.PropertyAttributes.InputSpecification("Select or provide a variable from the variable list")]
-        [Attributes.PropertyAttributes.SampleUsage("**vSomeVariable**")]
-        [Attributes.PropertyAttributes.Remarks("If you have enabled the setting **Create Missing Variables at Runtime** then you are not required to pre-define your variables, however, it is highly recommended.")]
-        public string v_userVariableName { get; set; }
+        [PropertyDescription("Output MailItem Directory")]   
+        [InputSpecification("Enter or Select the path of the directory to store the messages in.")]
+        [SampleUsage(@"C:\temp\myfolder || {vFolderPath} || {ProjectPath}\myFolder")]
+        [Remarks("This input is optional and will only be used if *SaveMailItems and Attachments* is set to *Yes*.")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Outlook Commands/GetOutlookEmailsCommand.cs]: Please add a space like **Save MailItems and Attachments** to refer to the above property description.
Apply same change to **"Output Attachment Directory"** property below.

File: taskt/Core/Automation/Commands/Outlook Commands/MoveCopyOutlookEmailsCommand.cs [hunk 2]
```diff
@@ -1,77 +1,78 @@
...
         {
-            this.CommandName = "MoveOutlookEmailsCommand";
-            this.SelectionName = "Move/Copy Outlook Emails";
-            this.CommandEnabled = true;
-            this.CustomRendering = true;
+            CommandName = "MoveCopyOutlookEmailsCommand";
+            SelectionName = "Move/Copy Outlook Emails";
+            CommandEnabled = true;
+            CustomRendering = true;
+            v_OperationType = "Move Emails";
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Outlook Commands/MoveCopyOutlookEmailsCommand.cs:70]: Also initialize **v_MoveCopyUnreadOnly** like `v_MoveCopyUnreadOnly = "Yes"`

File: taskt/Core/Automation/Commands/Outlook Commands/MoveCopyOutlookEmailsCommand.cs [hunk 3]
```diff
@@ -141,18 +135,19 @@ public override void RunCommand(object sender)
...
+            RenderedControls.AddRange(CommandControls.CreateDefaultDropdownGroupFor("v_OperationType", this, editor));
+            RenderedControls.AddRange(CommandControls.CreateDefaultDropdownGroupFor("v_MoveCopyUnreadOnly", this, editor));
 
             return RenderedControls;
         }
 
         public override string GetDisplayValue()
         {
-            return base.GetDisplayValue() + $" [From: {v_SourceFolder}, To: {v_DestinationFolder}]";
+            return base.GetDisplayValue() + $" [From '{v_SourceFolder}' - Filter by '{v_Filter}' - {v_OperationType} Emails to '{v_DestinationFolder}']";
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Outlook Commands/MoveCopyOutlookEmailsCommand.cs]: Please change it to `$" [{v_OperationType} to '{v_DestinationFolder}' From '{v_SourceFolder}' - Filter by '{v_Filter}']"`

File: taskt/Core/Automation/Commands/Outlook Commands/SendOutlookEmailCommand.cs [hunk 4]
```diff
@@ -1,93 +1,89 @@
...
     {
         [XmlAttribute]
-        [Attributes.PropertyAttributes.PropertyDescription("Please Indicate Recipients (; delimited)")]
-        [Attributes.PropertyAttributes.InputSpecification("Enter the Email Addresses of the recipients in semicolon seperated values")]
-        [Attributes.PropertyAttributes.SampleUsage("test@test.com;test2@test.com")]
-        [Attributes.PropertyAttributes.Remarks("")]
-        [Attributes.PropertyAttributes.PropertyUIHelper(Attributes.PropertyAttributes.PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
+        [PropertyDescription("Recipient(s)")]
+        [InputSpecification("Enter the email address(es) of the recipient(s).")]
+        [SampleUsage("test@test.com;test2@test.com || {vEmail1};{vEmail2} || {vEmails}")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Outlook Commands/SendOutlookEmailCommand.cs]: alsoprovide a sample for a single recipient

File: taskt/Core/Automation/Commands/Outlook Commands/SendOutlookEmailCommand.cs [hunk 5]
```diff
@@ -1,93 +1,89 @@
...
-        public string v_Attachment { get; set; }
-
-        [XmlAttribute]
-        [Attributes.PropertyAttributes.PropertyDescription("Provide Email Subject")]
-        [Attributes.PropertyAttributes.InputSpecification("Enter the subject you want sent in your Email")]
-        [Attributes.PropertyAttributes.SampleUsage("**myData**")]
-        [Attributes.PropertyAttributes.Remarks("")]
-        [Attributes.PropertyAttributes.PropertyUIHelper(Attributes.PropertyAttributes.PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
+        [PropertyDescription("Email Subject")]
+        [InputSpecification("Enter the subject you want sent in your Email")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Outlook Commands/SendOutlookEmailCommand.cs]: You can simply write "Enter the subject of the email."
Punctuation is also missing.

File: taskt/Core/Automation/Commands/Outlook Commands/SendOutlookEmailCommand.cs [hunk 6]
```diff
@@ -1,93 +1,89 @@
...
 
         public override void RunCommand(object sender)
         {
 
             var vRecipients = v_Recipients.ConvertToUserVariable(sender);
-            var vAttachment = v_Attachment.ConvertToUserVariable(sender);
+            var vAttachment = v_Attachments.ConvertToUserVariable(sender);
             var vSubject = v_Subject.ConvertToUserVariable(sender);
             var vBody = v_Body.ConvertToUserVariable(sender);
             var vBodyType = v_BodyType.ConvertToUserVariable(sender);
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Outlook Commands/SendOutlookEmailCommand.cs]: `v_BodyType` doesn't need conversion

File: taskt/Core/Automation/Commands/Outlook Commands/GetOutlookEmailsCommand.cs [hunk 7]
```diff
@@ -1,109 +1,115 @@
...
 
 namespace taskt.Core.Automation.Commands
 {
     [Serializable]
-    [Attributes.ClassAttributes.Group("Outlook Commands")]
-    [Attributes.ClassAttributes.Description("This command allows you to get emails and attachments with outlook")]
-    [Attributes.ClassAttributes.UsesDescription("Use this command when you want to get emails and attachments with your currenty logged in outlook account")]
-    [Attributes.ClassAttributes.ImplementationDescription("")]
+    [Group("Outlook Commands")]
+    [Description("This command gets selected emails and their attachments in Outlook.")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Outlook Commands/GetOutlookEmailsCommand.cs]: change to "This command gets selected emails and their attachments **from** Outlook."

**Review by ob-ab — CHANGES_REQUESTED** [review]
Please make a few changes.

File: taskt/Core/Automation/Commands/Outlook Commands/GetOutlookEmailsCommand.cs (hunk 1 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Outlook Commands/GetOutlookEmailsCommand.cs]: done

File: taskt/Core/Automation/Commands/Outlook Commands/MoveCopyOutlookEmailsCommand.cs (hunk 2 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Outlook Commands/MoveCopyOutlookEmailsCommand.cs:70]: done

File: taskt/Core/Automation/Commands/Outlook Commands/MoveCopyOutlookEmailsCommand.cs (hunk 3 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Outlook Commands/MoveCopyOutlookEmailsCommand.cs]: done

File: taskt/Core/Automation/Commands/Outlook Commands/SendOutlookEmailCommand.cs (hunk 4 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Outlook Commands/SendOutlookEmailCommand.cs]: done

File: taskt/Core/Automation/Commands/Outlook Commands/SendOutlookEmailCommand.cs (hunk 5 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Outlook Commands/SendOutlookEmailCommand.cs]: done

File: taskt/Core/Automation/Commands/Outlook Commands/SendOutlookEmailCommand.cs (hunk 6 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Outlook Commands/SendOutlookEmailCommand.cs]: done

File: taskt/Core/Automation/Commands/Outlook Commands/GetOutlookEmailsCommand.cs (hunk 7 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Outlook Commands/GetOutlookEmailsCommand.cs]: done

**Review by ob-ab — APPROVED** [review]
Looks good to me now.

**Review by CameronHerwig — APPROVED** [review]

**Pull request merged**